src/equations/polytropic_euler_2d.jl

```diff
@@ -0,0 +1,172 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+    @doc raw"""
+        PolytropicEulerEquations2D(gamma, kappa)
+
+    The polytropic Euler equations
+    ```math
+    \partial t
+    \begin{pmatrix}
+    \rho \\ \rho v_1 \\ \rho v_2
+    \end{pmatrix}
+    +
+    \partial x
+    \begin{pmatrix}
+     \rho v_1 \\ \rho v_1^2 + \kappa\rho^\gamma \\ \rho v_1 v_2
+    \end{pmatrix}
+    +
+    \partial y
+    \begin{pmatrix}
+    \rho v_2 \\ \rho v_1 v_2 \\ \rho v_2^2 + \kappa\rho^\gamma
+    \end{pmatrix}
+    =
+    \begin{pmatrix}
+    0 \\ 0 \\ 0
+    \end{pmatrix}
+    ```
+    for an ideal gas with ratio of specific heats `gamma`
+    in two space dimensions.
+    Here, ``\rho`` is the density and ``v_1`` and`v_2` the velocities and
+    ```math
+    p = \kappa\rho^\gamma
+    ```
+    the pressure, which we replaced using this relation.
+
+    """
+    struct PolytropicEulerEquations2D{RealT <: Real, RealT <: Real} <:
+           AbstractPolytropicEulerEquations{2, 3}
+        gamma::RealT               # ratio of specific heats
+        kappa::RealT               # fluid scaling factor
+        inv_gamma_minus_one::RealT # = inv(gamma - 1); can be used to write slow divisions as fast multiplications
+
+        function PolytropicEulerEquations2D(gamma, kappa)
+            new{typeof(gamma), typeof(kappa)}(gamma, kappa)
+        end
+    end
+
+    function varnames(::typeof(cons2cons), ::PolytropicEulerEquations2D)
+        ("rho", "rho_v1", "rho_v2")
+    end
+    varnames(::typeof(cons2prim), ::PolytropicEulerEquations2D) = ("rho", "v1", "v2")
+
+    # Calculate 1D flux for a single point in the normal direction
+    # Note, this directional vector is not normalized
+    @inline function flux(u, normal_direction::AbstractVector,
+                          equations::PolytropicEulerEquations2D)
+        rho_e = last(u)
+        rho, v1, v2 = cons2prim(u, equations)
+        p = pressure(u, equations)
+
+        v_normal = v1 * normal_direction[1] + v2 * normal_direction[2]
+        rho_v_normal = rho * v_normal
+        f1 = rho_v_normal
+        f2 = rho_v_normal * v1 + p * normal_direction[1]
+        f3 = rho_v_normal * v2 + p * normal_direction[2]
+        return SVector(f1, f2, f3)
+    end
+
+    @inline function flux_ranocha(u_ll, u_rr, normal_direction::AbstractVector,
+                                  equations::PolytropicEulerEquations2D)
```

Comment:
Have you verified that this is still a "true" Ranocha flux, i.e., that it preserves entropy etc.?

Reply:
As discussed elsewhere, maybe just remove the flux for now. Alternatively, you could try and implement the EC flux from https://doi.org/10.1007/s10543-019-00789-w

Reply:
I took the liberty to implement an (untested) version of the generic flux needed for either isothermal or polytropic gases. The entropy variables defined in the current implementation were also wrong so I implemented those too. Finally, I implemented the new special mean value function that is called `stolarsky_mean`, although this name could be changed.

To avoid an overly long post (plus I could not figure out how make a suggestion in the unaltered file `src/auxiliary/math.jl`) I attached the three new functions in a file. Github complained that I could not upload a file with a `.jl` extension so it is `.txt`.

[polytropic_scratch.txt](https://github.com/trixi-framework/Trixi.jl/files/11944839/polytropic_scratch.txt)

Reply:
Thanks @andrewwinters5000 . I will give this a try.

Reply:
@andrewwinters5000 , your implementation seems to work nicely. But still have to test EC.

Reply:
That is good, I assume you tested convergence (?). You can adapt one of the weak blast wave tests from the compressible Euler 2D file to test the entropy conservation.

Reply:
Code converges now.